pkg/rhoda/rhoda_test.go

```diff
@@ -18,165 +22,379 @@ import (
+							Namespace: namespace,
+							Name:      string(provider.SecretData["providerType"]),
+						},
+						DBaaSInventorySpec: dbaasv1alpha1.DBaaSInventorySpec{
+							CredentialsRef: &dbaasv1alpha1.NamespacedName{
+								Namespace: namespace,
+								Name:      provider.SecretName,
+							},
+						},
+					},
+				}
+				err = client.Create(context.Background(), &inventory)
+				Expect(err).NotTo(HaveOccurred())
 
-		Context("Create Inventory", func() {
-			It("Create inventory using the map", func() {
-				for _, value := range providers {
-					fmt.Println(value)
+				//sleep timer to make sure things run in parallel
+				//fmt.Printf("Current Unix Time: %v\n", time.Now())
+				//time.Sleep(30 * time.Second)
+				//fmt.Printf("Current Unix Time: %v\n", time.Now())
 
-					scheme := runtime.NewScheme()
-					err := dbaasv1alpha1.AddToScheme(scheme)
+				//Check inventories status
+				inventory = dbaasv1alpha1.DBaaSInventory{}
+				Eventually(func() bool {
+					fmt.Println("Checking status for : " + provider.ProviderName)
+					err := client.Get(context.Background(), k8sClient.ObjectKey{
+						Namespace: namespace,
+						Name:      "provider-acct-test-e2e-" + provider.ProviderName,
+					}, &inventory)
 					Expect(err).NotTo(HaveOccurred())
+					if len(inventory.Status.Conditions) > 0 {
+						return inventory.Status.Conditions[0].Type == "SpecSynced" && inventory.Status.Conditions[0].Status == "True"
+					} else {
+						fmt.Println("inventory.Status.Conditions Len is 0")
+						return false
+					}
+				}, 60*time.Second, 5*time.Second).Should(BeTrue())
 
-					//c, err := client.New(config, client.Options{Scheme: scheme})
-					//Expect(err).NotTo(HaveOccurred())
-
-					////create inventory
-					//inventory := dbaasv1alpha1.DBaaSInventory{
-					//	TypeMeta: meta.TypeMeta{
-					//		Kind:       "DBaaSInventory",
-					//		APIVersion: "dbaas.redhat.com/v1alpha1",
-					//	},
-					//	ObjectMeta: meta.ObjectMeta{
-					//		Name:      "provider-acct-test-e2e-" + providerName,
-					//		Namespace: namespace,
-					//		Labels:    map[string]string{"related-to": "dbaas-operator", "type": "dbaas-vendor-service"},
-					//	},
-					//	Spec: dbaasv1alpha1.DBaaSOperatorInventorySpec{
-					//		ProviderRef: dbaasv1alpha1.NamespacedName{
-					//			Namespace: namespace,
-					//			Name:      string(secretData["providerType"]),
-					//		},
-					//		DBaaSInventorySpec: dbaasv1alpha1.DBaaSInventorySpec{
-					//			CredentialsRef: &dbaasv1alpha1.NamespacedName{
-					//				Namespace: namespace,
-					//				Name:      "dbaas-test-e2e-" + providerName,
-					//			},
-					//		},
-					//	},
-					//}
-					//if err = c.Create(context.Background(), &inventory); err != nil {
-					//	fmt.Printf("Failed to create invenotry for : %v\n", err)
-					//}
-					fmt.Println("Get Inventory")
+				//test connection
+				if inventory.Status.Conditions[0].Type == "SpecSynced" && inventory.Status.Conditions[0].Status == "True" {
```

Comment:
What if there is a more recent status?

Reply:
Also as previously asked, do we reach this stage if Eventually fails? Do we need to check again?